src/SignalR/server/Core/src/Internal/HubMethodDescriptor.cs

```diff
@@ -276,14 +292,14 @@ private static Type GetServiceType(Type type)
         return type;
     }
 
-    private Type ValidateStreamType(Type streamType)
+    private Type ValidateParameterStreamType(Type streamType, Type parameterType)
     {
         if (!RuntimeFeature.IsDynamicCodeSupported && streamType.IsValueType)
         {
-            // NativeAOT apps are not able to stream IAsyncEnumerable and ChannelReader of ValueTypes
-            // since we cannot create AsyncEnumerableAdapters.MakeCancelableAsyncEnumerator and AsyncEnumerableAdapters.MakeAsyncEnumeratorFromChannel methods with a generic ValueType.
+            // NativeAOT apps are not able to stream IAsyncEnumerable and ChannelReader of ValueTypes as parameters
+            // since we cannot create a concrete IAsyncEnumerable and ChannelReader of ValueType to pass into the Hub method.
             var methodInfo = MethodExecutor.MethodInfo;
-            throw new InvalidOperationException($"Unable to stream an item with type '{streamType}' on method '{methodInfo.DeclaringType}.{methodInfo.Name}' because it is a ValueType. Native code to support streaming this ValueType will not be available with native AOT.");
+            throw new InvalidOperationException($"Method '{methodInfo.DeclaringType}.{methodInfo.Name}' is not supported with native AOT because it has a parameter of type '{parameterType}'. Streaming parameters of ValueTypes is not supported because the native code to support the ValueType will not be available with native AOT.");
```

Comment:
I think singular is correct here because the action "Streaming" is the subject. "Stream _(things)_ is not supported".

* Herding cows is fun.
vs.
* Herding cows are fun.

Reply:
IMO: "ValueType streaming parameters are not supported because..."

Reply:
Thanks. Updated.